src/Playwright/Core/Artifact.cs

```diff
@@ -67,7 +67,7 @@ public async Task SaveAsAsync(string path)
             await _channel.SaveAsAsync(path).ConfigureAwait(false);
             return;
         }
-        System.IO.Directory.CreateDirectory(Path.GetDirectoryName(path));
+        System.IO.Directory.CreateDirectory(Path.GetDirectoryName(Path.GetFullPath(path)));
```

Comment:
Does System.IO.Directory.CreateDirectory not work with relative paths?

Reply:
It does, but the case I was fixing it was just foo.zip